packages/flutter/lib/src/material/search.dart

```diff
@@ -327,6 +328,12 @@ class _SearchPageRoute<T> extends PageRoute<void> {
     super.didComplete(result);
     assert(delegate._route == this);
     delegate._route = null;
+    // Ensure that [showSearch] future completes when the search route is
+    // removed by other means than calling [SearchDelegate.close].
+    if (!delegate._result.isCompleted) {
+      delegate._result.complete(null);
```

Comment:
why is this not completing with "result"?

Reply:
or even better, why is it not just the route's future?

Reply:
I refactored it to use the route's future

(In a previous iteration we weren't able to do that because the search page consisted of two page routes, but that's no longer true and the completer was just a no-longer-needed artifact of that implementation)